src/tox/tox_env/python/api.py

```diff
@@ -4,19 +4,28 @@
 from __future__ import annotations
 
 import logging
+import re
 import sys
 from abc import ABC, abstractmethod
 from pathlib import Path
 from typing import Any, List, NamedTuple, cast
 
-from packaging.tags import INTERPRETER_SHORT_NAMES
 from virtualenv.discovery.py_spec import PythonSpec
 
 from tox.config.main import Config
 from tox.tox_env.api import ToxEnv, ToxEnvCreateArgs
 from tox.tox_env.errors import Fail, Recreate, Skip
 from tox.util.ci import is_ci
 
+PY_FACTORS_RE = re.compile(
+    r"""
+    ^(?!py$)                                               # don't match 'py' as it doesn't provide any info
```

Comment:
This is taken from tox 3 code. It confused me too initially but in hindsight it makes sense. This will mean we ignore `py` but not e.g. `py3` or `py311`. The idea being that `py` doesn't actually mean anything other than "use some type of Python"...which we're obviously going to do because it's `tox`.

Reply:
Yeah, figured it out eventually.